src/authentication/authentication-hapi-plugin-spec.ts

```diff
@@ -305,6 +313,33 @@ describe('authentication-hapi-plugin', () => {
       const team = JSON.parse(response.payload);
       expect(team['invitation-token']).to.be.undefined;
     });
+
+    it('should not leak information from GitLab', async () => {
+      // Arrange
+      const callerTeamId = 1;
+      const { server, plugin } = await getServer(
+        (p: AuthenticationHapiPlugin) => [
+          sinon.stub(p, '_getUserGroups')
+            .returns(Promise.resolve([{
+              id: callerTeamId,
+              name: 'foo',
+              web_url: 'http://test.com',
+              visibility_level: 0,
+            }])),
+        ],
+      );
+
+      // Act
+      const response = await makeRequest(server, `/team`);
+
+      // Assert
+      expect(response.statusCode).to.eq(200);
+      expect(plugin._getUserGroups).to.have.been.calledOnce;
+      const team = JSON.parse(response.payload);
+      expect(team.id).to.eq(callerTeamId);
+      expect(team.web_url).to.not.exist;
+      expect(team.visibility_level).to.not.exist;
```

Comment:
The first test tests for the existence of properties that we want to include. The point of that test is to just make sure that extra stuff is not leaked, so I'd not duplicate the test here. We could even remove the `expect(team.id).to.eq(callerTeamId);` test here.